### ISSUE TITLE
Search only searching in challenge spec and not title

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/challenge-api#390]
@rootelement , when we search for a specific term, challenge api returns only when search term is found in the challenge description. It does not consider title. This was working fine before.

Example:
<img width="1440" alt="Screenshot 2021-02-23 at 10 51 07 AM" src="https://user-images.githubusercontent.com/58783823/108805080-1cf8b300-75c5-11eb-86f8-c80a60cc5408.png">
<img width="1440" alt="Screenshot 2021-02-23 at 10 51 23 AM" src="https://user-images.githubusercontent.com/58783823/108805086-21bd6700-75c5-11eb-9433-ba908f1379a6.png">

**Comment by ThomasKranitsas** [comment]
@SathyaJayabal in order to fix/improve our search functionality, can you please create a list of expected search queries and results along with a simple explanation so we can improve our code? Something like:
- Searching for `node API` should return `Implement an API using NodeJs`. Explanation: all query terms are partially present in any order

We need as many use case scenarios as possible so we build a strong searching mechanism.

**Comment by rootelement** [comment]
Might be a question for @Oanh-and-only-Oanh as well.

**Comment by ThomasKranitsas** [comment]
@SathyaJayabal @Oanh-and-only-Oanh any update here?

**Comment by SathyaJayabal** [comment]
@ThomasKranitsas , the basic search which was working before is :
Searching for `node API` should return challenges which have the words `node API` either in the title or the description(contest spec).

@Oanh-and-only-Oanh , please add any further requirements.

**Comment by ThomasKranitsas** [comment]
IMHO, searching for node API should return challenges that:

1. have the `node API` in the title (case insensitive)
2. OR have the `node API`  in the description (case insensitive)
3. OR have the tag `node` (partial match - case insensitive)
4. OR have the tag `API` (partial match - case insensitive)

we could make this more advanced by extracting all words in #1 and use them separately as a query with lower weight so we can also match things like Implement an API using NodeJS  (both terms are present but in a different order)

Let me know if we should implement the search this way

**Comment by rootelement** [comment]
I almost think the weight should be:
1. Title Exact Match Phrase
2. Tag exact match
3. Description Exact Match
4. Title Fuzzy Match (tokenized)
5. Tag Fuzzy
6. Description Fuzzy

**Comment by Oanh-and-only-Oanh** [comment]
I agree with @rootelement but as a minimum we'd like it to return challenges which has the search terms either in the title or the description(contest spec).

**Comment by Oanh-and-only-Oanh** [comment]
@rootelement @ThomasKranitsas, when will this fixed be released?

**Comment by ThomasKranitsas** [comment]
@Oanh-and-only-Oanh there's a PR https://github.com/topcoder-platform/challenge-api/pull/428 for this.

@rootelement should we merge that to prod?

**Comment by rootelement** [comment]
This was pushed to prod earlier today

**Comment by SathyaJayabal** [comment]
@ThomasKranitsas, still search does not return the challenge when we search by a word in the challenge title
<img width="1440" alt="Screenshot 2021-07-16 at 9 42 55 AM" src="https://user-images.githubusercontent.com/58783823/125891096-233f3e07-9fbf-49c6-9c41-1826bd76a666.png">
<img width="1440" alt="Screenshot 2021-07-16 at 9 43 16 AM" src="https://user-images.githubusercontent.com/58783823/125891109-abfc8f8e-a1ae-479e-80e8-1569a0a0b1b6.png">

**Comment by SathyaJayabal** [comment]
@ThomasKranitsas , search is now not returning when the search term is found in a tag. 

For example, when we click on any of the tags for a challenge in listings page, it doesn't return any results.
<img width="1440" alt="Screenshot 2021-09-28 at 11 52 32 AM" src="https://user-images.githubusercontent.com/58783823/135034035-026f1d86-c320-482b-8587-4630eee8a117.png">

**Comment by SathyaJayabal** [comment]
cc @luizrrodrigues
